masonry_core/src/core/widget_pod.rs

```diff
@@ -13,89 +15,120 @@ pub struct WidgetPod<W: ?Sized> {
     inner: WidgetPodInner<W>,
 }
 
+/// A container for a widget yet to be inserted.
+#[non_exhaustive]
+#[expect(missing_docs, reason = "Names are self-explanatory.")]
+pub struct NewWidget<W: ?Sized> {
+    pub widget: Box<W>,
+    pub id: WidgetId,
+    pub options: WidgetOptions,
+    pub properties: Properties,
+}
+
+/// The options a new widget will be created with.
+#[derive(Default, Debug)]
+pub struct WidgetOptions {
+    /// The transform the widget will be created with.
+    pub transform: Affine,
+    /// The disabled state the widget will be created with.
+    pub disabled: bool,
+}
+
 // TODO - This is a simple state machine that lets users create WidgetPods
```

Comment:
Is this whole comment still true? It wasn't clear if it was referring to the bunch of stuff following it or just the `CreateWidget` struct which you're removing.

Reply:
It's still true. The "state machine" is specifically the enum that follows.